File: SudokuKata/SudokuKataTests/ConsoleOutput.cs
using System;
using System.IO;

namespace SudokuKataTests
{
    public partial class ProgramTests
    {
        // From http://www.vtrifonov.com/2012/11/getting-console-output-within-unit-test.html
        public class ConsoleOutput : IDisposable
        {
            private StringWriter stringWriter;
            private TextWriter originalOutput;

            public ConsoleOutput()
            {
                stringWriter = new StringWriter();
                originalOutput = Console.Out;
                Console.SetOut(stringWriter);
            }

            public string GetOuput()
            {
                return stringWriter.ToString();
            }

            public void Dispose()
            {
                Console.SetOut(originalOutput);
                stringWriter.Dispose();
            }
        }
    }
}

File: SudokuKata/SudokuKataTestsCpp/TestHelpers/ConsoleOutput.cpp
#include "ConsoleOutput.h"

namespace SudokuKataTests::TestHelpers
{

	ConsoleUtilities::ConsoleOutput::ConsoleOutput()
	{
		stringWriter = new StringWriter();
		originalOutput = Console::Out;
		Console::SetOut(stringWriter);
	}

	std::wstring ConsoleUtilities::ConsoleOutput::GetOuput()
	{
//C# TO C++ CONVERTER TODO TASK: There is no C++ equivalent to 'ToString':
		return stringWriter->ToString();
	}

	void ConsoleUtilities::ConsoleOutput::Dispose()
	{
		delete originalOutput;
		Console::SetOut(originalOutput);
		stringWriter->Dispose();
	}
}
